### PR TITLE
Adds PipelineContext to be passed to execution state. 

### PR DESCRIPTION
**Pull request by nitinmotgi**

### CODE REVIEW

File: src/main/java/co/cask/wrangler/steps/Lower.java [hunk 1]
```diff
@@ -37,11 +38,12 @@ public Lower(int lineno, String detail, String col) {
    * Transforms a column value from any case to lower case.
    *
    * @param row Input {@link Row} to be wrangled by this step.
+   * @param context Specifes the context of the pipeline.
```
Comment (russorat) [inline src/main/java/co/cask/wrangler/steps/Lower.java]: Specifies

File: src/test/java/co/cask/wrangler/steps/PipelineTest.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public void testBasicPipelineWorking() throws Exception {
 
     // Run through the wrangling steps.
     for (Step step : steps) {
-      row = (Row) step.execute(row);
+      row = (Row) step.execute(row,null);
```
Comment (russorat) [inline src/test/java/co/cask/wrangler/steps/PipelineTest.java]: checkstyle

**Comment by russorat** [comment]
spelling and checkstyle, otherwise, LGTM

**Comment by nitinmotgi** [comment]
@russorat addressed your comments.

**Comment by nitinmotgi** [comment]
```
addressed your comments

[10:21]  
can I merge ?

Russ Savage [10:21] 
yep

N1Tin m0Tg1 [10:23] 
thanks
```

**Pull request merged**